activerecord/lib/active_record/errors.rb

```diff
@@ -178,18 +178,10 @@ class Rollback < ActiveRecordError
   class DangerousAttributeError < ActiveRecordError
   end
 
-  # Raised when unknown attributes are supplied via mass assignment.
-  class UnknownAttributeError < NoMethodError
-
-    attr_reader :record, :attribute
-
-    def initialize(record, attribute)
-      @record = record
-      @attribute = attribute.to_s
-      super("unknown attribute '#{attribute}' for #{@record.class}.")
-    end
-
-  end
+  UnknownAttributeError = ActiveSupport::Deprecation::DeprecatedConstantProxy.new( # :nodoc:
+    'ActiveRecord::UnknownAttributeError', 
+    'ActiveModel::AttributeAssignment::UnknownAttributeError'
```

Comment:
Given that end users may be rescuing from this error in their apps, it seems like a rather long and specific namespace. Perhaps keep it under `ActiveModel::UnknownAttributeError`?